react/components/ResourceList/index.js

```diff
@@ -5,36 +5,175 @@ import Table from '../Table'
 import Pagination from '../Pagination'
 import InputSearch from '../InputSearch'
 import Button from '../Button'
+import Toggle from '../Toggle'
 import IconDownload from '../icon/Download'
 import IconUpload from '../icon/Upload'
+import IconColumns from '../icon/Columns'
+const MAX_FIELDS_BOX_HEIGHT = 192
 
 class ResourceList extends PureComponent {
+  constructor(props) {
+    super(props)
+    this.state = {
+      clonedSchema: props.table.schema ? this.cloneSchema(props.table.schema) : {},
+      showFieldsOptions: false,
+    }
+  }
+
+  cloneSchema = (schema, showAll = false) => {
+    const clonedSchema = { properties: {} }
+    Object.keys(schema.properties).forEach(key => {
+      if (!schema.properties[key].hidden || showAll) {
+        clonedSchema.properties[key] = { ...schema.properties[key] }
```

Comment:
isso aqui (fazendo `...`) só faz um clone de um nível; se o schema tiver mais de um nível de objetos (e.g. `{a: {b: ...}}`), tu só vai estar copiando as referencias dos objetos; ou seja, tu vai estar podendo modificar os objetos originais, e/ou mudanças nos originais vao afetar o teu clone. 

se só for um nivel mesmo nao tem problema, mas se tiver mais é melhor talvez usar uma funçao pronta de clone deep.